DHCP is mandatory when using wifi_nxp

### Describe the bug

If I want to use `CONFIG_WIFI_NXP=y` in my project configuration, I must also define `CONFIG_NET_DHCPV4=y` to be able to build, even if I want to setup a static IP configuration.

I'm using Zephyr 4.3.0 through the NXP downstream repo (https://github.com/nxp-zephyr/nxp-zephyr-repo).

### Regression

- [ ] This is a regression.

### Steps to reproduce

_No response_

### Relevant log output

```shell
[0/1] Re-running CMake...Loading Zephyr default modules (Zephyr base (cached)).
-- Application: C:/Users/A98032420/test_7/XXX/apps/wifi_cli
-- CMake version: 4.2.1
-- Cache files will be written to: C:/Users/A98032420/test_7/zephyr/.cache
-- Zephyr version: 4.3.0 (C:/Users/A98032420/test_7/zephyr)
-- Found west (found suitable version "1.5.0", minimum required is "0.14.0")
-- Board: ubx_evk_iris_w1, Revision: fidelex, qualifiers: rw612
-- Found host-tools: zephyr 0.17.4 (C:/Users/A98032420/zephyr-sdk-0.17.4)
-- Found toolchain: zephyr 0.17.4 (C:/Users/A98032420/zephyr-sdk-0.17.4)
-- Found BOARD.dts: C:/Users/A98032420/test_7/zephyr/boards/u-blox/ubx_evk_iris_w1/ubx_evk_iris_w1_rw612.dts
-- Generated zephyr.dts: C:/Users/A98032420/test_7/XXX/apps/wifi_cli/build/zephyr/zephyr.dts
-- Generated pickled edt: C:/Users/A98032420/test_7/XXX/apps/wifi_cli/build/zephyr/edt.pickle
-- Generated devicetree_generated.h: C:/Users/A98032420/test_7/XXX/apps/wifi_cli/build/zephyr/include/generated/zephyr/devicetree_generated.h
Parsing C:/Users/A98032420/test_7/zephyr/Kconfig
Loaded configuration 'C:/Users/A98032420/test_7/zephyr/boards/u-blox/ubx_evk_iris_w1/ubx_evk_iris_w1_rw612_fidelex_defconfig'
Merged configuration 'C:/Users/A98032420/test_7/XXX/apps/wifi_cli/prj.conf'
Configuration saved to 'C:/Users/A98032420/test_7/XXX/apps/wifi_cli/build/zephyr/.config'
Kconfig header saved to 'C:/Users/A98032420/test_7/XXX/apps/wifi_cli/build/zephyr/include/generated/zephyr/autoconf.h'
-- Looking for device RW612 in C:/Users/A98032420/test_7/modules/hal/nxp/mcux/mcux-sdk-ng/devices/
-- Found device folder: C:/Users/A98032420/test_7/modules/hal/nxp/mcux/mcux-sdk-ng/devices/Wireless/RW/RW612
--  generate include of binary blob: C:/Users/A98032420/test_7/modules/hal/nxp/zephyr/blobs/rw61x/rw61x_sb_wifi_a2.bin
els_pkc module is included.
Loading Zephyr default modules (Zephyr base (cached)).
-- Found gen_kobject_list: C:/Users/A98032420/test_7/zephyr/scripts/build/gen_kobject_list.py
-- Configuring done (18.5s)
-- Generating done (1.8s)
-- Build files have been written to: C:/Users/A98032420/test_7/XXX/apps/wifi_cli/build

[160/262] Building C object modules/hal_nxp/CMakeFiles/modules...les/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c.ob
C:/Users/A98032420/test_7/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c: In function 'stop_cb':
C:/Users/A98032420/test_7/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c:995:5: warning: implicit declaration of function 'net_dhcpv4_stop' [-Wimplicit-function-declaration]
  995 |     net_dhcpv4_stop(if_handle->netif);
      |     ^~~~~~~~~~~~~~~
C:/Users/A98032420/test_7/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c: In function 'net_configure_address':
C:/Users/A98032420/test_7/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c:1155:13: warning: implicit declaration of function 'net_dhcpv4_restart' [-Wimplicit-function-declaration]
 1155 |             net_dhcpv4_restart(if_handle->netif);
      |             ^~~~~~~~~~~~~~~~~~
[256/262] Linking CXX executable zephyr\zephyr_pre0.elf; Logical command for additional byproducts on target: zephyr_pre0
FAILED: [code=1] zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map C:/Users/A98032420/test_7/XXX/apps/wifi_cli/build/zephyr/zephyr_pre0.map
C:\WINDOWS\system32\cmd.exe /C "cd . && C:\Users\A98032420\zephyr-sdk-0.17.4\arm-zephyr-eabi\bin\arm-zephyr-eabi-g++.exe -DUSE_RTOS=1 -DUSE_RTOS=1 -DUSE_RTOS=1 -DUSE_RTOS=1 -DOSA_USED -DUSE_RTOS=1 -DMFLASH_FILE_BASEADDR=7340032 -gdwarf-4 -Os zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr\zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/arm/core/offsets/offsets.c.obj  -T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map,C:/Users/A98032420/test_7/XXX/apps/wifi_cli/build/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/libarch__arm__core.a  zephyr/arch/arch/arm/core/cortex_m/libarch__arm__core__cortex_m.a  zephyr/arch/arch/arm/core/cortex_m/cmse/libarch__arm__core__cortex_m__cmse.a  zephyr/arch/arch/arm/core/mpu/libarch__arm__core__mpu.a  zephyr/lib/libc/picolibc/liblib__libc__picolibc.a  zephyr/lib/libc/common/liblib__libc__common.a  zephyr/lib/posix/c_lib_ext/liblib__posix__c_lib_ext.a  zephyr/lib/net_buf/liblib__net_buf.a  zephyr/subsys/random/libsubsys__random.a  zephyr/subsys/net/l2/ethernet/libsubsys__net__l2__ethernet.a  zephyr/subsys/net/l2/wifi/libsubsys__net__l2__wifi.a  zephyr/subsys/net/ip/libsubsys__net__ip.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/entropy/libdrivers__entropy.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/pinctrl/libdrivers__pinctrl.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  zephyr/drivers/wifi/libdrivers__wifi.a  modules/hal_nxp/libmodules__hal_nxp.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -LC:/Users/A98032420/test_7/XXX/apps/wifi_cli/build/zephyr  zephyr/arch/common/libisr_tables.a  -fuse-ld=bfd  -mcpu=cortex-m33+nodsp  -mthumb  -mabi=aapcs  -mfp16-format=ieee  -mtp=soft  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -Wl,-no-pie  -Wl,--undefined=_sw_isr_table  -Wl,--undefined=_irq_vector_table  -specs=picolibc.specs  -DPICOLIBC_LONG_LONG_PRINTF_SCANF -Lc:/users/a98032420/zephyr-sdk-0.17.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/thumb/v8-m.main/nofp  -lc -lgcc && C:\WINDOWS\system32\cmd.exe /C "cd /D C:\Users\A98032420\test_7\XXX\apps\wifi_cli\build\zephyr && "C:\Program Files\CMake\bin\cmake.exe" -E true""
c:/users/a98032420/zephyr-sdk-0.17.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd.exe: modules/hal_nxp/libmodules__hal_nxp.a(net.c.obj): in function `stop_cb':
C:/Users/A98032420/test_7/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c:995: undefined reference to `net_dhcpv4_stop'
c:/users/a98032420/zephyr-sdk-0.17.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd.exe: modules/hal_nxp/libmodules__hal_nxp.a(net.c.obj): in function `net_interface_dhcp_stop':
C:/Users/A98032420/test_7/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c:1042: undefined reference to `net_dhcpv4_stop'
c:/users/a98032420/zephyr-sdk-0.17.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd.exe: modules/hal_nxp/libmodules__hal_nxp.a(net.c.obj): in function `net_configure_address':
C:/Users/A98032420/test_7/modules/hal/nxp/mcux/middleware/wifi_nxp/port/net/zephyr/net.c:1155: undefined reference to `net_dhcpv4_restart'
collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

_No response_

### Additional Context

_No response_

--- a/west.yml
+++ b/west.yml
@@ -210,7 +210,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: 2f8883b3358e7be2101075cd0f00dd2e0fa68050
+      revision: af5d95d3e6be13f1b993d2c466595e7cc71ba57b
       path: modules/hal/nxp
       groups:
         - hal

